proposals/Scoped-Custom-Element-Registries.md

```diff
@@ -134,68 +115,24 @@ Another option for looking up registries is to store an element's originating re
 
-In the other direction, we want to be able to call `new MyElement()`, which means we need to be able to locate the correct tag name from a constructor as well.
-
-The way this is done is by limiting constructors by default to only looking up registrations from the global registry. If the constructor is not defined in the global registry, it will throw. In order to get a constructor that creates a scoped definition, customElementRegistry.define() returns a new constructor:
-
-```js
-import {OtherElement} from './my-element.js';
-
-const registry = new CustomElementRegistry();
-
-// define() returns a new class:
-const LocalOtherElement = registry.define('other-element-2', OtherElement);
-const el = new LocalOtherElement();
-el.tagName === 'other-element-2';
-
-// The same class is available from registry.get():
-const O = registry.get('other-element-2')
-const el2 = new O();
-el2.tagName === 'other-element-2';
-```
-
-The constructor returned by `define()` is from a trivial subclass of the registered class.
-
-### Sugar for bulk registrations
-
-To make importing and registering multiple definitions easier, the CustomElementRegistry constructor can take an object containing multiple definitions:
+As a result, it must limit constructors by default to only looking up registrations from the global registry. If the constructor is not defined in the global registry, it will throw.
 
-```js
-import {ElementOne} from './element-one.js';
-import {ElementTwo} from './element-two.js';
-
-const registry = new CustomElementRegistry({
-  definitions: {
-    'element-one': ElementOne,
-    'element-two': ElementTwo,
-  }
-});
-
-export class MyElement extends HTMLElement {
-  constructor() {
-    this.attachShadow({mode: 'open', registry});
-  }
-}
-```
+This poses a limitation for authors trying to use the constructor to create new elements associated to scoped registries but not registered as global. More flexibility can be analyzed post MVP, for now, a user-land abstraction can help by keeping track of the constructor and its respective registry.
 
 ## API
 
 ### CustomElementRegistry
 
 #### Constructor
 
-The CustomElementRegistry constructor creates a new instance of CustomElementRegistry, independent of the instance available at `window.customElements`.
+The `CustomElementRegistry` constructor creates a new instance of CustomElementRegistry, independent of the instance available at `window.customElements`.
 
 ##### Syntax
 
 ```js
-const registry = new CustomElementRegistry({parent, definitions});
+const registry = new CustomElementRegistry();
 ```
 
-##### Parameters
-parent A parent CustomElementRegistry to inherit definitions from.
-definitions An object whose keys are custom element names and values are the associated constructors.
```

Comment:
I don't recall any movement or consensus on removing the bulk registration constructor parameter. I don't see any in the minutes.

Reply:
From what I recall, we said that this was another thing that could be done in user-land, and since this was supposed to work in tandem with `getDefinitions`, which was removed, it seems fine to be removed as well.

Reply:
I dont remember any decision on this either.

Reply:
Alright, it might be just me, maybe bias here! If it is not in the notes, and nobody remembers about it, I can bring it back :)

Reply:
I really think we should make the ergonomics nice here. "can be done in userland" shouldn't be used to make the built-in API require more boilerplate than necessary. If this were a point of contention, then removing it could be on the path to consensus, but I didn't hear implementors object to it. The minutes don't have anything either.

Reply:
fair enough, I will update it in a bit.

Reply:
Ok, let me provide more details, to see how to bring this back:

1. since `parent` is gone, adding one options bag with a single option seems weird, in the language spec, we usually reject that, not sure what we do about it in the DOM.
2. should only own names be used? or inheritance is supported there? I suspect it should be only own names to be registered (I don't remember any precedent about this in the DOM APIs).
3. what should happen if an invalid name is provided? as today, `ce.define()` will throw for invalid names. Should the registry throw during construction? should it ignore them? This seems important because it adds another point of failure.

Any feedback here? You probably see why taking this out was the easiest path for MVP.

Reply:
@justinfagnani @LarsDenBakker any comments on the issues described here?

Reply:
Personally I like the bulk registration, with scoped registration this will be very common. But it can be done in userland too, and if there are complications in how to spec this or if this is going to delay this proposal in any way I'm fine with dropping it.

Reply:
Hi! I'm jumping in this whole topic and I've been reading the changes about it for a couple days. 

There is an interesting case here where I'd personally hold on the bulk registration. I also understand the advantages for it and - as a web practitioner - I'd totally go for this extra.

For me, it seems like we could experiment going on the max/min practice here, right? So it seems if we ship `new CustomElementsRegistry` without args we could expand it later. That gives room to see how userland adopts it and finally expand better if necessary.

For one side, what I see is:

```js
const registry = new CustomElementsRegistry({
  definitions: {
    'foo': Foo,
    'bar': Bar,
  }
});
```

as a short for

```js
const registry = new CustomElementsRegistry();

registry.define('foo', Foo);
registry.define('bar', Bar);
```

I believe this code will be often abstracted, the benefit is not so big for this piece of feature.

For sure, consuming the return of `getDefinitions` would increase its benefit here, but this is now yet happening and we could eventually define something that resolves this matter. The options parameter would set a requirement before we can spec it.

---

There is another separation here which I appreciate for the low level api: the name validation. As @caridy [has mentioned](https://github.com/justinfagnani/webcomponents/pull/1#discussion_r407625204), we need to handle the name validation. Once again, as a web practitioner I'd rather have the steps separated to handle everything step by step.

This also brings me to another idea, if I may, where the CustomElementRegistry class only allow one definition per time. So I suppose this definitions options should be part of a general discussion for Web Components having a `.define()` variation for multiple definitions as well. I'm not saying we should or shouldn't spec this method, but I believe the way @Caridy has proposed here matches the current API design.

I'm sorry if my comment is too long or repeating what other people has said. I'm new here but I'm pretty interested in the matters and I hope we find a good conclusion in the end.